src/server/api/routes/create-batch.router.js

```diff
@@ -3,13 +3,23 @@ const express = require('express');
 
 const router = express.Router({ mergeParams: true });
 
+const {
+  checkIfAuthenticated,
+} = require('../lib/middleware/authentication.middleware');
+
 // controllers
 const createBatchController = require('../controllers/create-batch.controller');
 
 // ENDPOINT: /api/modules/:cropId :GET to get one crop's default values
-router.get('/:cropId', (req, res, next) => {
+router.post('/create-batch', checkIfAuthenticated, (req, res, next) => {
```

Comment:
the middleware for authorization should be used
checkIfAuthorized(ROLES.SUPER_ADMIN,ROLE.ADMIN)
since only admin and super_admin can add a batch